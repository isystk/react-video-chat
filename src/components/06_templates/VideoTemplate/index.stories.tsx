import { Meta, Story } from '@storybook/react'
import React from 'react'
import VideoTemplate from './index'
import MainService, { Member } from '@/services/main'
import ChanelService from '@/services/Chanel'

export default {
  title: '06_templates/VideoTemplate',
  component: VideoTemplate,
} as Meta

const Template: Story = () => {
  const main = new MainService(() => ({}))
  main.setName('isystk')
  main.setRoomId('test')
  main.addChanel(
    new ChanelService(
      main,
      'all',
      'すべて',
      'all',
      'images/friends/Alpha_Team.png',
      'ルーム内のすべてのメンバー'
    )
  )
  main.addMember({
    connectionId: 'aaa',
    name: 'aaa',
    photo: 'images/friends/BigBoss.png',
  } as Member)
  main.setChanelId('all')
  main.video.isPeerConnected = true
  main.video.members = [
    {
      connectionId: 'aaa',
      name: 'aaa',
      photo: 'images/friends/BigBoss.png',
    } as Member,
  ]
  const props: VideoTemplate = {main}
  return (
    <VideoTemplate {...props}/>
  )
}

export const Primary = Template.bind({})
Primary.storyName = 'プライマリ'
Primary.args = {}
