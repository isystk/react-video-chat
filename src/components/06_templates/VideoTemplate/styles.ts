import { css, keyframes } from '@emotion/css'

const useStyles = (theme) => css`{
  root: {
    flexGrow: 1,
  },
  paper: {
    padding: theme.spacing(2),
    textAlign: 'center',
    color: theme.palette.text.secondary,
  },
}`

export { useStyles }
