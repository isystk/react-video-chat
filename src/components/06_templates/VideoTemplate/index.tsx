import Grid from '@mui/material/Grid'
import React, { FC, useContext } from 'react'
import VideoLocal from '../../04_organisms/VideoLocal'
import VideoRemote from '../../04_organisms/VideoRemote'
import { ContainerProps, WithChildren } from 'types'
import { useStyles } from './styles'
import { connect } from '@/components/hoc'
import HtmlSkeleton, { Title } from '@/components/05_layouts/HtmlSkeleton'

/** VideoTemplateProps Props */
export type VideoTemplateProps = WithChildren & { main }
/** Presenter Props */
export type PresenterProps = VideoTemplateProps & { classes; title; grid }

/** Presenter Component */
const VideoTemplatePresenter: FC<PresenterProps> = ({
  main,
  classes,
  title,
  grid,
  ...props
}) => (
  <HtmlSkeleton>
    <Title>{title}</Title>
    <div className={classes.root}>
      <Grid container spacing={0}>
        <Grid item {...grid}>
          <VideoLocal />
        </Grid>
        {main.video.members.map(function (member, idx) {
          return (
            <Grid item {...grid} key={idx}>
              <VideoRemote member={member} />
            </Grid>
          )
        })}
      </Grid>
    </div>
  </HtmlSkeleton>
)

/** Container Component */
const VideoTemplateContainer: React.FC<
  ContainerProps<VideoTemplateProps, PresenterProps>
> = ({ presenter, children, main, ...props }) => {
  const classes = useStyles()
  const grids = {
    0: { xs: 12, sm: 6, md: 6 },
    1: { xs: 12, sm: 6, md: 6 },
    2: { xs: 12, sm: 6, md: 6 },
    3: { xs: 12, sm: 6, md: 6 },
    4: { xs: 12, sm: 4, md: 4 },
    5: { xs: 12, sm: 4, md: 4 },
    6: { xs: 12, sm: 3, md: 3 },
    7: { xs: 12, sm: 3, md: 3 },
  }
  const grid = grids[Math.min(main.video.members.length, 7)]

  if (!main.video.isPeerConnected) return <></>
  return presenter({
    children,
    main,
    classes,
    title: main.room.name,
    grid,
    ...props,
  })
}

export default connect<VideoTemplateProps, PresenterProps>(
  'VideoTemplate',
  VideoTemplatePresenter,
  VideoTemplateContainer
)
