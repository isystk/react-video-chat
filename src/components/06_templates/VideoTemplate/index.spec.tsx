import React from 'react'
import renderer from 'react-test-renderer'
import VideoTemplate, {VideoTemplateProps} from './index'
import '@testing-library/jest-dom/extend-expect'
import MainService, { Member } from '@/services/main'
import ChanelService from '@/services/Chanel'
import {applyMiddleware, createStore} from "redux";
import thunk from "redux-thunk";
import { Provider } from 'react-redux'
import reducers from '@/stores'

describe('VideoTemplate', () => {
  it('Match Snapshot', () => {
    const enhancer = applyMiddleware(thunk)
    const store = createStore(reducers, enhancer)
    const main = new MainService(() => ({}))
    main.setName('isystk')
    main.setRoomId('test')
    main.addChanel(
      new ChanelService(
        main,
        'all',
        'すべて',
        'all',
        'images/friends/Alpha_Team.png',
        'ルーム内のすべてのメンバー'
      )
    )
    main.addMember({
      connectionId: 'aaa',
      name: 'aaa',
      photo: 'images/friends/BigBoss.png',
    } as Member)
    main.setChanelId('all')
    main.video.isPeerConnected = true
    main.video.members = [
      {
        connectionId: 'aaa',
        name: 'aaa',
        photo: 'images/friends/BigBoss.png',
      } as Member,
    ]

    const props: VideoTemplateProps = {main}
    const component = renderer.create(
      <Provider store={store}>
        <VideoTemplate {...props} />
      </Provider>
    )
    const tree = component.toJSON()

    expect(tree).toMatchSnapshot()
  })
})
