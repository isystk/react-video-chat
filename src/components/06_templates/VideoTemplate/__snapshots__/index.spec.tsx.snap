// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`VideoTemplate Match Snapshot 1`] = `null`;
